gnocchi/storage/_carbonara.py

```diff
@@ -117,9 +118,15 @@ def _store_metric_measures(metric, timestamp_key, aggregation,
                                granularity, data, offset=None, version=3):
         raise NotImplementedError
 
-    @staticmethod
-    def _list_split_keys_for_metric(metric, aggregation, granularity,
+    def _list_split_keys_for_metric(self, metric, aggregation, granularity,
                                     version=3):
+        return set(map(
+            functools.partial(carbonara.SplitKey, sampling=granularity),
+            utils.to_timestamps(self._list_split_keys(
```

Comment:
the following is my understanding.

`+    def _list_split_keys(self, metric, aggregation, granularity, version=3):
          key = self._metric_key(metric)
          if not self._client.exists(key):
              raise storage.MetricDoesNotExist(metric)
         split_keys = set()
         hashes = self._client.hscan_iter(
             key, match=self._aggregated_field_for_split(aggregation, '*',
                                                         granularity, version))
         for f, __ in hashes:
             meta = f.decode("utf8").split(self.FIELD_SEP, 1)
             split_keys.add(meta[0])
         return split_keys`

this seems to be just returning a set of strings representing epoch time in seconds. if we pass it string to splitkey, it will correctly convert it to epoch time in seconds cast to float: https://github.com/gnocchixyz/gnocchi/blob/master/gnocchi/carbonara.py#L411

it seems to_timestamps unnecessarily converts to nanosecond float, https://github.com/gnocchixyz/gnocchi/blob/master/gnocchi/utils.py#L129... maybe i'm missing something.